src/main/java/com/joyent/manta/cosbench/MantaStorage.java

```diff
@@ -209,15 +266,24 @@ public void createContainer(final String container, final Config config) {
     @Override
     public void deleteContainer(final String container, final Config config) {
         if (logging) {
-            logger.info("Performing DELETE at /{}", container);
+            if ("buckets".equals(testType)) {
+                logger.info("Performing DELETE bucket at /{}", container);
+            } else {
+                logger.info("Performing DELETE dir at /{}", container);
+            }
         }
 
         try {
-            client.deleteRecursive(directoryOfContainer(container));
+            if ("buckets".equals(testType)) {
+                client.deleteBucket(pathOfBaseContainer(container));
```

Comment:
If there are still objects in the bucket when it tries to delete it, we should proactively handle that error.

Reply:
Added a fall back mechanism to empty the bucket and then retry to delete it.